packages/vc-verification/test/chain-of-trust-test.ts

```diff
@@ -0,0 +1,454 @@
+          roleDefinition: {
+            roleName: activeuserRole,
+            enrolmentPreconditions: [
+              { type: PreconditionType.Role, conditions: [userRole] },
+            ],
+            requestorFields: [],
+            issuerFields: [],
+            issuer: { issuerType: 'ROLE', roleName: managerRole },
+            revoker: { revokerType: 'ROLE', roleName: managerRole },
+            metadata: [],
+            roleType: '',
+            version: defaultVersion,
+          },
+        }),
+      })
+    ).wait();
+
+    await (
+      await deployer.sendTransaction({
+        ...roleFactory.newRole({
+          domain: managerRole,
+          roleDefinition: {
+            roleName: managerRole,
+            enrolmentPreconditions: [],
+            requestorFields: [],
+            issuerFields: [],
+            issuer: { issuerType: 'ROLE', roleName: adminRole },
+            revoker: { revokerType: 'ROLE', roleName: adminRole },
+            metadata: [],
+            roleType: '',
+            version: defaultVersion,
+          },
+        }),
+      })
+    ).wait();
+  });
+
+  describe('chainOfTrustTests', () => {
+    it('verifies hierarchy of issuers', async () => {
+      const claim: OffChainClaim = {
+        claimTypeVersion: 1,
+        issuedToken: 'token',
+        iss: adminDid,
+        claimType: adminRole,
+      };
+
+      let token: string = '';
+      let ipfsCID: string = '';
+      if (admin.privateKey) {
+        token = jwt.sign(claim, admin.privateKey, { algorithm: 'ES256' });
+
+        if (token) {
+          ipfsCID = await store.save(token);
+        }
+      }
+      const serviceId = adminRole;
+      const updateData: IUpdateData = {
+        type: DIDAttribute.ServicePoint,
+        value: {
+          id: `${adminDid}#service-${serviceId}`,
+          type: 'ClaimStore',
+          serviceEndpoint: ipfsCID,
+        },
+      };
+      await adminOperator.update(
+        adminDid,
+        DIDAttribute.ServicePoint,
+        updateData,
+        validity
+      );
+      const vc: IVerifiableCredential = {
+        '@context': [],
+        id: adminDid,
+        type: ['Claims'],
+        issuer: adminDid,
+        issaunceDate: '02/02/2022',
+        credentialSubject: {
+          id: adminDid,
+          role: {
+            namespace: adminRole,
+            version: '1',
+          },
+          issuerFields: [],
+        },
+        proof: {
+          '@context': 'string',
+          verificationMethod: 'string',
+          created: 'string',
+          proofPurpose: 'string',
+          type: 'string',
+        },
+      };
+
+      const credentialUrl = await store.save(token);
+      await adminOperator.update(adminDid, DIDAttribute.ServicePoint, {
+        type: DIDAttribute.ServicePoint,
+        value: {
+          id: adminDid,
+          type: adminRole,
+          serviceEndpoint: credentialUrl,
+        },
+      });
+
+      console.log('helllllooooo');
```

Comment:
Hello! :wave:

Reply:
Just for debugging, tests aren't finished yet.

Reply:
I know, I know. I just couldn't resist :smile: